ArcGISRuntimeSDKQt_QMLSamples/Routing/NavigateRoute/NavigateRoute.qml

```diff
@@ -0,0 +1,277 @@
+import Esri.ArcGISRuntime 100.8
+import QtQuick.Layouts 1.11
+import QtPositioning 5.2
+import Esri.samples 1.0
+
+Rectangle {
+    id: rootRectangle
+    clip: true
+    width: 800
+    height: 600
+
+    readonly property url routeTaskUrl: "https://sampleserver6.arcgisonline.com/arcgis/rest/services/NetworkAnalysis/SanDiego/NAServer/Route"
+    property var m_route: null
+    property var m_routeResult: null
+    property var directionListModel: null
+    property string textString: ""
+
+    MapView {
+        id: mapView
+        anchors.fill: parent
+
+        Map {
+            BasemapNavigationVector {}
+        }
+
+        GraphicsOverlay {
+            id: routeOverlay
+            Graphic {
+                id: routeAheadGraphic
+                SimpleLineSymbol {
+                    style: Enums.SimpleLineSymbolStyleDash
+                    color: "blue"
+                    width: 5
+                }
+            }
+            Graphic {
+                id: routeTraveledGraphic
+                SimpleLineSymbol {
+                    style: Enums.SimpleLineSymbolStyleSolid
+                    color: "cyan"
+                    width: 3
+                }
+            }
+            Graphic {
+                Point {
+                    id: conventionCenterPoint
+                    x: -117.160386727
+                    y: 32.706608
+                    SpatialReference {wkid: 4326}
+                }
+                symbol: stopSymbol
+            }
+            Graphic {
+                Point {
+                    id: memorialPoint
+                    x: -117.173034
+                    y: 32.712327
+                    SpatialReference {wkid: 4326}
+                }
+                symbol: stopSymbol
+            }
+            Graphic {
+                Point {
+                    id: aerospaceMuseumPoint
+                    x: -117.147230
+                    y: 32.730467
+                    SpatialReference {wkid: 4326}
+                }
+                symbol: stopSymbol
+            }
+        }
+
+        SimpleMarkerSymbol {
+            id: stopSymbol
+            style: "SimpleMarkerSymbolStyleDiamond"
+            color: "red"
+            size: 20
+        }
+
+        Stop {
+            id: stop1
+            geometry: conventionCenterPoint
+        }
+        Stop {
+            id: stop2
+            geometry: memorialPoint
+        }
+        Stop {
+            id: stop3
+            geometry: aerospaceMuseumPoint
+        }
+
+        RouteTask {
+            id: routeTask
+            url: routeTaskUrl
+            Component.onCompleted: {
+                load();
+            }
+            onLoadStatusChanged: {
+                if (loadStatus === Enums.LoadStatusLoaded) {
+                    createDefaultParameters();
+                }
+            }
+            onCreateDefaultParametersStatusChanged: {
+                if (createDefaultParametersStatus === Enums.TaskStatusCompleted) {
```

Comment:
I find this kind of thing more readable if you early out instead because everything is less indented:

```suggestion
                if (createDefaultParametersStatus != Enums.TaskStatusCompleted)
                    return;
```